model/role.go

```diff
@@ -160,6 +160,20 @@ func ChannelModeratedPermissionsChangedByPatch(role *Role, patch *RolePatch) []s
 	return result
 }
 
+// GetChannelModeratedPermissions returns a map of channel moderated permissions that the role has access to
+func (r *Role) GetChannelModeratedPermissions() map[string]bool {
+	moderatedPermissions := make(map[string]bool)
```

Comment:
I created [a helper function for doing this](https://github.com/mattermost/mattermost-server/pull/13813/files#diff-6f76ad473b02b813b08e80eeb2ab681cR16). I have a TODO to migrate instances in Mattermost to use it, which I'll do in a PR separate from our Channel Moderation code.